code/modules/food/cooking_machines.dm

```diff
@@ -477,6 +477,88 @@ var/global/ingredientLimit = 10
 	takeIngredient(I, L, TRUE) //shove the item in, even if it can't be deepfried normally
 	empty_icon()
 
+
+// Sugarizer ///////////////////////////////////////
+// its like a deepfrier
+
+// but with sugar
+
+#define SUGARIZER_MINSUGAR 50
+
+/obj/machinery/cooking/deepfryer/sugarizer
+	name = "sugarizer"
+	desc = "Creates sugar copies of stuff."
+	icon_state = "sugarizer_off"
+	icon_state_on = "sugarizer_on"
+	foodChoices = null
+	cookTime = 100
+	recursive_ingredients = 1
+	cks_max_volume = 400
+	cooks_in_reagents = 1
+
+/obj/machinery/cooking/deepfryer/sugarizer/validateIngredient(var/obj/item/I, var/force_cook)
+	if(I.w_class < W_CLASS_LARGE)
+		. = "valid"
+
+	else
+		. = "The sugarizer will not be able to replicate that."
+	if((. == "valid") && (!foodNesting))
+		if(findtext(I.name,"sugar"))
```

Comment:
Here's a fun fact: the deep fryer does this same thing where it will refuse to fry anything with "fried" in its name. Like the deep fryer, you'll be able to bypass this just by renaming your food with a pen. So if you sugarize a cookie, it becomes a "sugar cookie", smack it with a pen and name it anything else and you can resugarize it over and over.

What I would probably do in this case is just check if there's "too much sugar" in it already or something.